docs/DeveloperGuide.md

```diff
@@ -149,103 +149,104 @@ Classes used by multiple components are in the `seedu.storemando.commons` packag
 
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+#### Clear Feature
 
-</div>
+##### Implementation
 
-The `redo` command does the opposite — it calls `Model#redoStoreMando()`, which shifts the `currentStatePointer` once to
-the right, pointing to the previously undone state, and restores the inventory to that state.
+This portion describes the implementation of the clear feature which allows users to clear items that are either in
+a particular location or clear all items in the list.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `storeMandoStateList.size() - 1`, pointing to the latest inventory state, then there are no undone StoreMando states to restore. The `redo` command uses `Model#canRedoStoreMando()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+1. When the user keys in a command string, `execute` command of the `LogicManager` is called with the given string as input.
+2. In the method, `LogicManager` calls on the `parseCommand` method of StoreMandoParser to parse the given command.
+3. `StoreMandoParser` parses the string and determines the command given is a `ClearCommand`.
+    Then, a `ClearCommandParser` object is created to further parse the command.
+4. `StoreMandoParser` then calls on the `parse` method of `ClearCommandParser` to parse the arguments provided.
+5. `ClearCommandParser` checks if the argument provided is an empty string. If so, the constructor of `ClearCommand` 
+    without any parameters is called. Else, a `LocationContainsPredicate` object will be created with the arguments 
+    as parameter.`CommandParseException` will be thrown if this is not possible.
+6. `ClearCommand` object will be returned to `ClearCommandParser` which then returns it to `LogicManager`. `LogicManager`
+    then calls the `execute` method of `ClearCommand` with a `Model` object, model, as parameter.
+7. `ClearCommand` calls on the `clearLocation` method of model with the `ClearCommand`'s attribute predicate as parameter. 
+    Subsequently, it calls on model's `updateFilteredItemList`.
+8. Finally, a `CommandResult` object is created and is returned to `LogicManager`.   
 
-</div>
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the inventory, such as `list`,
-will usually not call `Model#commitStoreMando()`, `Model#undoStoreMando()` or `Model#redoStoreMando()`. Thus,
-the `storeMandoStateList` remains unchanged.
+The following sequence diagram shows how clear feature works:
 
-![UndoRedoState4](images/UndoRedoState4.png)
+![ClearSequenceDiagram](images/ClearSequenceDiagram.png)
```

Comment:
One suggestion as mentioned above is to include the sequence diagram above the implementation.